packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -48,6 +48,25 @@ export interface RoleProps {
    */
   readonly inlinePolicies?: { [name: string]: PolicyDocument };
 
+  /**
+   * AWS supports permissions boundaries for IAM entities (users or roles).
+   * A permissions boundary is an advanced feature for using a managed policy
+   * to set the maximum permissions that an identity-based policy can grant to
+   * an IAM entity. An entity's permissions boundary allows it to perform only
+   * the actions that are allowed by both its identity-based policies and its
+   * permissions boundaries.
+   *
+   * You can use an AWS managed policy or a customer managed policy to set the
+   * boundary for an IAM entity (user or role). That policy limits the maximum
+   * permissions for the user or role.
+   *
+   * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-permissionsboundary
+   * @link https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html
+   *
+   * @default - No permissions boundaries required.
+   */
+  readonly permissionsBoundary?: IPolicy;
```

Comment:
Pretty sure this needs to be a managed policy instead of a policy. AFAIK you can not attach an inline policy as a permissions boundary. Once this proposed change is through (https://github.com/aws/aws-cdk/issues/2974#issuecomment-517526071), customer managed policies will then also be supported.

Reply:
I created a PR for that change here https://github.com/aws/aws-cdk/pull/3578 so people can get some eyes on it